### ISSUE TITLE
Suggested improvements to beer song exercise

### ISSUE THREAD

**Issue by kytrinyx** [exercism/elixir#385]
Related to the [99 Bottles of OOP](https://www.sandimetz.com/99bottles/) book, we have a repository containing all the available BeerSong exercises from Exercism.

One reader, @dcarral, submitted a whole host of little fixes and improvements, that I'd like to bring into Exercism where appropriate.

The changes relevant to the Elixir track are:

> * Remove implementation of methods' signatures in Elixir (failing test is enough to drive their construction).

The changes are in this commit: https://github.com/dcarral/99bottles-polyglot/pull/11/commits/21d6d25aaf170c303ab5382d2d35fdf027ee332c

Since the BeerSong exercise is pretty far down the list in the Elixir track, I think it's probably safe to remove the scaffolding, provided that the failing test is enough to drive their construction, as indicated in @dcarral's change.

A further question is whether or not the scaffolding should be removed from all but the first N exercises, if the test failures can drive the method construction.

**Comment by devonestes** [comment]
Yeah, I guess technically we could remove the scaffolding for folks. The failing tests would indeed guide them as to what they would need to write.

I'm not really sure if removing those bits would provide all that much of a better experience for users, though. I especially like that we include typespecs and documentation in the scaffolding, which is a nice thing for folks to get used to seeing all the time.

So, I think the pros are that users would have to rely on their failing tests a little more, which is a good thing.

The cons are that we lose out on the ability to reinforce for people the best practice of writing documentation and typespecs.

Personally, I'm in favor of just keeping things as is, but I don't feel super strongly about it. If people think removing these empty function definitions would give users a better experience, then I wouldn't be just fine with doing that.

**Comment by kytrinyx** [comment]
> The cons are that we lose out on the ability to reinforce for people the best practice of writing documentation and typespecs.

I'd argue that if we are writing the documentation and typespecs, then we're probably not helping reinforce the habit for them. Maybe this is something we could reinforce with feedback?

I've noticed in the Go track that people don't get good habits around documentation unless they wrestle with it themselves a bit.

**Comment by kytrinyx** [comment]
This is one of the places where automated linting could help, too, potentially. For v2 we won't be giving automated feedback to the user, but rather provide automated hints to the mentors as "stuff to potentially talk about".

**Comment by workingjubilee** [comment]
student feedback thing:

I don't feel an urge to write typespecs in code so far. I have seen them a few times, so I know what they look like, but in spite of also dabbling with C & Rust, typespecs in Elixir don't... motivate me, actually.

I do feel a desire to write missing docstrings. I've started doing that.

Silly exercise idea for progressively/optionally/mostly-inferred type langs: midway through, an exercise which is just "write typespecs and docs." "Dealing with an egregious coworker" might come off wrong? Perhaps "you came back to your code three months later and have no idea what it does, document it so you don't have to remember next time" would be more #relatable. The tests would be a (highly opinionated) linter.
